src/framework/testing.ts

```diff
@@ -0,0 +1,66 @@
+import getPort from 'get-port'
+import { GraphQLClient } from 'graphql-request'
+import { App } from './app'
+import * as Layout from './layout'
+
+type AppClient = {
+  query: GraphQLClient['request']
+}
+
+export function createQueryClient(apiUrl: string): AppClient {
+  const client = new GraphQLClient(apiUrl)
+
+  return {
+    query(queryString, variables) {
+      return client.request(queryString, variables)
+    },
+  }
+}
+
+export type TestContext = {
+  app: AppClient
+  teardown: () => Promise<void>
+}
+
+/**
+ * Setup a test context providing utilities to query against your GraphQL API
+ */
+export async function setupTest(): Promise<TestContext> {
+  process.env.GRAPHQL_SANTA_STAGE = 'dev'
+
+  const port = await getPort({ port: getPort.makeRange(4000, 6000) })
+  const apiUrl = `http://localhost:${port}/graphql`
+
+  const app = require('./index').app as App
+
+  const oldServerStart = app.server.start
+
+  app.server.start = () => {
+    return oldServerStart({
+      port,
+      playground: false,
+      introspection: false,
+      startMessage: () => '',
+    })
+  }
+
+  const appModule = await Layout.findAppModule()
```

Comment:
hmmmm

Reply:
Curve ball... Lack of conventions, and config... reluctantly see no trivial improvement to suggest.

Reply:
While if we can avoid async creation we can avoid having to import the type, that is also something the jest etc. test integration patterns can achieve among other things.